Framework/Algorithms/src/CompareWorkspaces.cpp

```diff
@@ -1134,6 +1162,90 @@ void CompareWorkspaces::doPeaksComparison(PeaksWorkspace_sptr tws1,
   }
 }
 
+//------------------------------------------------------------------------------------------------
+void CompareWorkspaces::doLeanElasticPeaksComparison(
+    LeanElasticPeaksWorkspace_sptr tws1, LeanElasticPeaksWorkspace_sptr tws2) {
+  // Check some table-based stuff
+  if (tws1->getNumberPeaks() != tws2->getNumberPeaks()) {
+    recordMismatch("Mismatched number of rows.");
+    return;
+  }
+  if (tws1->columnCount() != tws2->columnCount()) {
+    recordMismatch("Mismatched number of columns.");
+    return;
+  }
+
+  // sort the workspaces before comparing
+  auto sortPeaks = createChildAlgorithm("SortPeaksWorkspace");
+  sortPeaks->setProperty("InputWorkspace", tws1);
+  sortPeaks->setProperty("ColumnNameToSortBy", "DSpacing");
+  sortPeaks->setProperty("SortAscending", true);
+  sortPeaks->executeAsChildAlg();
+  IPeaksWorkspace_sptr ipws1 = sortPeaks->getProperty("OutputWorkspace");
+
+  sortPeaks = createChildAlgorithm("SortPeaksWorkspace");
+  sortPeaks->setProperty("InputWorkspace", tws2);
+  sortPeaks->setProperty("ColumnNameToSortBy", "DSpacing");
+  sortPeaks->setProperty("SortAscending", true);
+  sortPeaks->executeAsChildAlg();
+  IPeaksWorkspace_sptr ipws2 = sortPeaks->getProperty("OutputWorkspace");
+
+  const double tolerance = getProperty("Tolerance");
+  for (int i = 0; i < ipws1->getNumberPeaks(); i++) {
+    for (size_t j = 0; j < ipws1->columnCount(); j++) {
+      std::shared_ptr<const API::Column> col = ipws1->getColumn(j);
+      std::string name = col->name();
+      double s1 = 0.0;
+      double s2 = 0.0;
+      if (name == "RunNumber") {
+        s1 = double(ipws1->getPeak(i).getRunNumber());
+        s2 = double(ipws2->getPeak(i).getRunNumber());
+      } else if (name == "h") {
+        s1 = ipws1->getPeak(i).getH();
+        s2 = ipws2->getPeak(i).getH();
+      } else if (name == "k") {
+        s1 = ipws1->getPeak(i).getK();
+        s2 = ipws2->getPeak(i).getK();
+      } else if (name == "l") {
+        s1 = ipws1->getPeak(i).getL();
+        s2 = ipws2->getPeak(i).getL();
+      } else if (name == "Wavelength") {
+        s1 = ipws1->getPeak(i).getWavelength();
+        s2 = ipws2->getPeak(i).getWavelength();
+      } else if (name == "Energy") {
+        s1 = ipws1->getPeak(i).getFinalEnergy();
+        s2 = ipws2->getPeak(i).getFinalEnergy();
+      } else if (name == "DSpacing") {
+        s1 = ipws1->getPeak(i).getDSpacing();
+        s2 = ipws2->getPeak(i).getDSpacing();
+      } else if (name == "Intens") {
+        s1 = ipws1->getPeak(i).getIntensity();
+        s2 = ipws2->getPeak(i).getIntensity();
+      } else if (name == "SigInt") {
+        s1 = ipws1->getPeak(i).getSigmaIntensity();
+        s2 = ipws2->getPeak(i).getSigmaIntensity();
+      } else if (name == "BinCount") {
+        s1 = ipws1->getPeak(i).getBinCount();
+        s2 = ipws2->getPeak(i).getBinCount();
+      } else if (name == "QLab") {
```

Comment:
Can we also check `QSample`, there are cases where, for example with four-circle, the `QLab` is almost identical for every peak but the `QSample` is different because the goniometer is different.

Reply:
Added `QSample` to checklist as suggested.